onnxruntime/core/framework/bfc_arena.cc

```diff
@@ -407,6 +418,85 @@ void BFCArena::Free(void* p) {
   }
 }
 
+Status BFCArena::OnRunEnd() {
+  if (shrink_on_every_run_) {
```

Comment:
The caller can decide whether to call OnRunEnd or not. Not sure why the arena needs to maintain this boolean state?

Reply:
Sure- with the changed interface, callers can choose to call Shrink() if they want. This will go away anyway with the changed interface.

Reply:
Took this down